### DISCUSSION TITLE
请教下PCL是怎样做到自动下载forge和fabric的

### DISCUSSION THREAD

**Discussion by liuhaoyu-China** [Meloong-Git/PCL#2009] [提问] [answered]
最近用python写了一个一键开服（虽然没有GUI，但是下载配置启动一条龙服务也用不到GUI），已经支持spigot/paper/mohist了，但是forge和fabric只能下载到安装程序，没法自动安装，网上也没有相关教程，想着服务端应该和客户端差不多，就来这里请教了。请多多指点，谢谢

**Comment by so1ve** [comment]
我不太懂 PCL2 是怎么做的，不过你可以参考另外一个实现：https://github.com/Voxelum/minecraft-launcher-core-node/blob/master/packages/installer 大概就是下载 jar 然后手动实现一下安装过程，把 jar 里头的文件解压出来，然后移动到正确的位置

**Reply by liuhaoyu-China** [reply, marked as answer]
找到解决方法了，就先下载安装程序，然后给他一个启动参数，像这样java -jar forge-1.19.4-45.1.0-installer.jar --installServer就行了
fabric更简单，https://fabricmc.net/use/server/ 下载直接默认启动就行了
当时就没动脑子😂I'm sorry

**Reply by so1ve** [reply]
也可以 :)
